service/backend/pchain/indexer/types.go

```diff
@@ -0,0 +1,36 @@
+package indexer
+
+import (
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/vms/platformvm/genesis"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+)
+
+type ParsedBlock struct {
+	BlockID   ids.ID    `json:"id"`
+	BlockType string    `json:"type"`
+	ParentID  ids.ID    `json:"parent"`
+	Timestamp int64     `json:"timestamp"`
+	Height    uint64    `json:"height"`
+	Txs       []*txs.Tx `json:"transactions"`
+	Proposer  `json:"proposer"`
+}
+
+type GenesisBlockData struct {
+	Message       string          `json:"message"`
+	InitialSupply uint64          `json:"initialSupply"`
+	UTXOs         []*genesis.UTXO `json:"utxos"`
+}
+
+type ParsedGenesisBlock struct {
+	ParsedBlock
+	GenesisBlockData `json:"data"`
+}
+
+type Proposer struct {
+	ID           ids.ID     `json:"id"`
```

Comment:
q: I see the proposer certificate is not considered here. Is it on purpose? Note that certificate are *not* displayed in every blocks (there won't be a proposer in case of blocks built past proposer window)

Reply:
Not sure about the certificate. We used the indexer parser code shared by @rajranjan0608 . I will work with him offline to address this.

Reply:
to be clear, I am not saying it's wrong. I just realized there was a "missing" field and asked about the choice.